tests/test_losses.py

```diff
@@ -124,6 +124,60 @@ def test_rigids_riemannian_loss_and_grad(self):
 
         # self.assertTrue(np.allclose(result_2, curved_expected_2))
 
+    def test_caffe_gradient_checker(self):
+
+        print('\n #####  GRADIENT CHECKER (CAFFE) ##### \n')
+
+        print('Setting Test Parameters: ') # Adjustable Parameters
+        dx = 0.01
+        threshold = 0.01
+        y_pred = np.array([-0.662919 , -1.29015 , 1.30245 , 1.25567 , 1.09228 , 0.153399 ]) #Regularized
+        y_true = np.array([0.0349858 , 0.562573 , -1.94181 , -0.173889 , 0.127527 , -0.19467 ]) #Regularized
+
+        print('Stepsize: ', str(dx))
```

Comment:
ditto, I don't think all these are necessary. If needed, just do assertions in the code.